lib/new_relic/agent/logging.rb

```diff
@@ -89,7 +89,14 @@ def add_key_value message, key, value
 
         def escape(message)
           message = message.to_s unless message.is_a?(String)
-          message = message.scrub(REPLACEMENT_CHAR) unless message.valid_encoding?
+          unless message.valid_encoding?
```

Comment:
As I mentioned in #863 this will not cover ASCII-8BIT encoding because it's still valid. 
```
"č".force_encoding(Encoding::ASCII_8BIT).valid_encoding?
=> true
```


Try adding this test case:
```
def test_to_replace_ascii_8bit_chars
  message = 'message with an ASCII-8BIT character'
  input = "#{message} #{"č".force_encoding(Encoding::ASCII_8BIT)}"
  expectation = "#{message} #{DecoratingFormatter::REPLACEMENT_CHAR}#{DecoratingFormatter::REPLACEMENT_CHAR}"
  logger = DecoratingLogger.new @output
  logger.info input
  assert_equal expectation, last_message['message']
end
```

This is intended behavior. It will replace ASCII-8BIT `č` with two replacement charachters because `č` consists of `"\xC4\x8D"` and each character is invalid in UTF-8 and should therefore be replaced.